rewrite-java-tck/src/main/java/org/openrewrite/java/tree/SwitchPatternMatchingTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.tree;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.MinimumJava21;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+@MinimumJava21
+class SwitchPatternMatchingTest implements RewriteTest {
+
+    @Test
+    void shouldParseJava21PatternSwitch() {
+        rewriteRun(
+          java(
+            //language=java
+            """
+              class Test {
+                  String formatterPatternSwitch(Object obj) {
+                      return switch (obj) {
+                          case Integer i -> String.format("int %d", i);
+                          case Long l    -> String.format("long %d", l);
+                          case Double d  -> String.format("double %f", d);
+                          case String s  -> String.format("String %s", s);
+                          default        -> obj.toString();
+                      };
+                  }
+              }
+              """
+          ));
+    }
+
+    @Test
+    void shouldSupportParsingNullSwitch() {
+        rewriteRun(
+          java(
+            //language=java
+            """
+              class Test {
+                void fooBarWithNull(String s) {
+                    switch (s) {
+                        case null -> System.out.println("Oops");
+                        case "Foo", "Bar" -> System.out.println("Great");
+                        default -> System.out.println("Ok");
+                    }
+                }
+              }
+              """
+          ));
+    }
+
+    @Test
+    void shouldParseJava21EnumSupportInSwitch() {
+        rewriteRun(
+          java(
+            //language=java
+            """
+              sealed interface Currency permits Coin {}
+              enum Coin implements Currency { HEADS, TAILS }
+              
+              class Test {
+                  void switchEnum(Coin c) {
```

Comment:
Should we perhaps switch on `Currency` here? Or why was that sealed interface included above?
```suggestion
                  void switchEnum(Currency c) {
```

Reply:
The example was taken from here https://openjdk.org/jeps/441, not entirely sure why the sealed interface was used. I'm assuming defining just interface coin with it's 2 values would suffice?